### PR TITLE
OTT-979 :: Refactor Floor code

### PR DESCRIPTION
**Pull request by pm-nikhil-vaidya**
# Description

Please add change description or link to ticket, docs, etc.

# Checklist:

- [ ] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [ ] JIRA number is added in the PR title and the commit message.
- [ ] Updated the `header-bidding` repo with appropiate commit id.
- [ ] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

File: exchange/floors.go [hunk 1]
```diff
@@ -85,40 +85,41 @@ func enforceFloorToBids(bidRequestWrapper *openrtb_ext.RequestWrapper, seatBids
...
+			}
+			rate, err := getCurrencyConversionRate(seatBid.Currency, reqImpCur, conversions)
+			if err != nil {
+				errMsg := fmt.Errorf("error in rate conversion from = %s to %s with bidder %s for impression id %s and bid id %s", seatBid.Currency, reqImpCur, bidderName, bid.Bid.ImpID, bid.Bid.ID)
+				glog.Errorf(errMsg.Error())
+				errs = append(errs, errMsg)
+				continue
+			}
 
+			bidPrice := rate * bid.Bid.Price
```
Comment (pm-jaydeep-mohite) [inline exchange/floors.go:107]: Move updateBidExtWithFloors(reqImp, bid, reqImpCur) here 
to update bid extension irrespective bid is selected or rejected

Comment (pm-nikhil-vaidya) [reply exchange/floors.go:107]: In the previous flow, we are updating bidExt when the valid bid is selected,
If we move this then the previous assumption will be invalid
can you please confirm again on this

Comment (pm-jaydeep-mohite) [reply exchange/floors.go:107]: We need to update bidExt irrespective bid is selected or rejected

Comment (pm-nikhil-vaidya) [reply exchange/floors.go:107]: Okay

Comment (pm-nikhil-vaidya) [reply exchange/floors.go:107]: Done

**Review by pm-jaydeep-mohite — APPROVED** [review]
LGTM

**Pull request merged**